pkg/operation/botanist/waiter.go

```diff
@@ -133,11 +135,20 @@ func (b *Botanist) WaitUntilKubeAPIServerReady(ctx context.Context) error {
 	})
 }
 
-// WaitUntilVPNConnectionExists waits until a port forward connection to the vpn-shoot pod in the kube-system
+// WaitUntilTunnelConnectionExists waits until a port forward connection to the vpn-shoot pod in the kube-system
 // namespace of the Shoot cluster can be established.
-func (b *Botanist) WaitUntilVPNConnectionExists(ctx context.Context) error {
+func (b *Botanist) WaitUntilTunnelConnectionExists(ctx context.Context) error {
+	// If kubernetes version is >= 1.18 deploy the proxy-agent, otherwise deploy the proxy server
+	kubernetesVersionGeq118, err := versionutils.CheckVersionMeetsConstraint(b.Shoot.Info.Spec.Kubernetes.Version, ">= 1.18")
+	if err != nil {
+		return err
+	}
+
 	return retry.UntilTimeout(ctx, 5*time.Second, 900*time.Second, func(ctx context.Context) (done bool, err error) {
-		return b.CheckVPNConnection(ctx, b.Logger)
+		if kubernetesVersionGeq118 {
+			return b.CheckTunnelConnection(ctx, b.Logger, common.KonnectivityTunnel)
+		}
+		return b.CheckTunnelConnection(ctx, b.Logger, common.VPNTunnel)
```

Comment:
Hm, the `CheckTunnelConnection` is a method on the botanist, hence, it can "compute"/"know" the name of the deployment (vpn-shoot/konnectivity-agent) to check itself, so no need to pass it as an argument?